### PR TITLE
[Merged by Bors] - feat: `Finset.sup s f = 0 ↔ ∀ i ∈ s, f i = 0`

### PR DESCRIPTION
**Pull request by YaelDillies**
From ForbiddenMatrix

---

I am somewhat surprised no file exists for this already

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
What is ForbiddenMatrix?

File: Mathlib/Algebra/Order/Monoid/Canonical/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import Mathlib.Data.Finset.Lattice
+
+/-!
+# Extra lemmas about canonically ordered monoids
+-/
+
+namespace Finset
+variable {ι α : Type*} [CanonicallyLinearOrderedAddCommMonoid α] {s : Finset ι} {f : ι → α}
+
+@[simp] lemma sup_eq_zero : s.sup f = 0 ↔ ∀ i ∈ s, f i = 0 := by simp [← bot_eq_zero']
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/Monoid/Canonical/Basic.lean:16]: Should we have `inf_eq_zero` with `Exists` as well?

**Comment by YaelDillies** [comment]
YaelDillies/ForbiddenMatrix is the repository

File: Mathlib/Algebra/Order/Monoid/Canonical/Basic.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Order/Monoid/Canonical/Basic.lean:16]: We don't have the analogous lemma for `bot` at the moment

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Pull request closed without merging**